packages/server-wallet/src/protocols/__test__/open-channel.test.ts

```diff
@@ -1,95 +1,210 @@
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it(`does nothing if I've already deposited`, async () => {
+      await testCase({participant: 0, statesHeld, totalFunds: 5, fundsToDeposit: 0});
+    });
+  });
+  describe(`as participant 1`, () => {
+    it(`submits the transaction if it is my turn`, async () => {
+      const o = await testCase({participant: 1, statesHeld, totalFunds: 5, fundsToDeposit: 5});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it(`submits a top-up if there's a partial deposit`, async () => {
+      const o = await testCase({participant: 1, statesHeld, totalFunds: 7, fundsToDeposit: 3});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it.skip(`does nothing if I've already deposited`, async () => {
+      // todo: this fails, as it sends the post-fund-setup (as it probably should)
+      await testCase({participant: 1, statesHeld, totalFunds: 10, fundsToDeposit: 0});
+    });
   });
-  expect(protocol(ps)).toMatchObject(action);
 });
 
-test.each`
-  supported           | latestSignedByMe    | latest              | funding      | cond
-  ${undefined}        | ${prefundState0}    | ${prefundState0}    | ${funded}    | ${'when I have signed the prefund state, but it is not supported'}
-  ${undefined}        | ${undefined}        | ${undefined}        | ${funded}    | ${'when there is no state'}
-  ${prefundState0}    | ${postfundState2}   | ${postfundState2}   | ${funded}    | ${'when the prefund state is supported, and I have signed the postfund state'}
-  ${altPrefundState0} | ${altPrefundState0} | ${altPrefundState0} | ${notFunded} | ${'when the prefund state is supported, and alice is second to deposit'}
-  ${postfundState2}   | ${postfundState2}   | ${postfundState2}   | ${funded}    | ${'when the first postfund state is supported'}
-`('takes no action $cond', ({supported, latest, latestSignedByMe, funding}) => {
-  const ps = applicationProtocolState({app: {supported, latest, latestSignedByMe, funding}});
-  expect(protocol(ps)).toBeUndefined();
+describe(`post-fund-setup phase`, () => {
+  describe(`as participant 0`, () => {
+    it(`signs the post-fund-setup if all funds are present`, async () => {
+      await testCase({participant: 0, statesHeld: [0, 1], totalFunds: 10, statesToSign: [2]});
+    });
+    it(`doesn't do anything if I've already signed`, async () => {
+      await testCase({participant: 0, statesHeld: [0, 1, 2], totalFunds: 10, statesToSign: []});
+    });
+    it(`signs the post-fund-setup (and completes objective), when my opponent has signed`, async () => {
+      await testCase({
+        participant: 0,
+        statesHeld: [0, 1, 3],
+        totalFunds: 10,
+        statesToSign: [2],
+        completesObj: true,
+      });
+      // todo: and marks the objective as complete
```

Comment:
you have many `todo`s here. I think @snario prefers them to be `TODO`s. 

When should these be todone?

Reply:
This one has actually been done ✂️